app/components/EditProfileForm.tsx

```diff
@@ -49,23 +54,32 @@ export default function EditProfileForm({userData}) {
 
     const data = await res.json();
     console.log(data)
+    const status = res.status
+
+    if (status == 200) {
+      router.push("/dashboard")
+    }
+    else {
+      setErrorMessage(data['body'])
+    }
   }
   
   return (
     <>
         <form onSubmit={onSubmitHandler} className='w-full'>
+              <p>{errorMessage}</p>
               <label htmlFor='profilePicture' className='font-semibold'>Profile Picture:</label>
               <input type='file' name='profilePicture' className='w-full mb-2'/>
               <label htmlFor='firstName' className='font-semibold'>First Name:</label>
-              <input type='text' placeholder = {firstName} name='firstName' className='w-full border border-gray-200 shadow-inner mb-2 py-1 px-2 rounded-md'/>   
+              <input type='text' defaultValue= {firstName} name='firstName' className='w-full border border-gray-200 shadow-inner mb-2 py-1 px-2 rounded-md'/>   
               <label htmlFor='lastName' className='font-semibold'>Last Name:</label>
```

Comment:
```suggestion
              <input type='text' defaultValue={firstName} name='firstName' className='w-full border border-gray-200 shadow-inner mb-2 py-1 px-2 rounded-md'/>   
```
Same for the rest of the statements below.